nni/compression/pytorch/compressor.py

```diff
@@ -559,7 +559,7 @@ def forward(self, *inputs):
                 self.module.weight = new_weight
             else:
                 new_weight = self.module.old_weight
-                self.module.weight.data = new_weight.data
+                self.module.weight = new_weight.data
```

Comment:
why remove `data`?

Reply:
Because it can cause another error during backward.
![image](https://user-images.githubusercontent.com/50486294/132319398-afc59ce6-03f1-472b-a705-17171b1cac86.png)

By the way, i submit this pr to fix the original error which
![image](https://user-images.githubusercontent.com/50486294/132318573-83a8fc2b-fb7b-4a20-9383-12c4f3362aa9.png)
is caused by tensor assigned to parameter since parameter's assignment is invalid. Currently, we resolve this problem by converting parameter to buffer [here](https://github.com/microsoft/nni/blob/c7324918ecbc09cbbfa46e74aa175399de6ea973/nni/algorithms/compression/pytorch/quantization/quantizers.py#L256). And the `self.module.weight = new_weight.data` can be kept, everything works fine right now.